sonic-xcvrd/xcvrd/xcvrd.py

```diff
@@ -997,6 +997,16 @@ def get_cmis_media_lanes_mask(self, api, appl, lport, subport):
 
         return media_lanes_mask
 
+    def is_appl_reconfigure_required(self, api, app_new):
```

Comment:
@AnoopKamath function should do what its supposed to do. as the function name suggest, it should return True/False but not do or change the datapath state inside this function. 

```

for lane in range(8):
   curr_appl = api.get_application(lane)
   if (curr_appl != 0) :
	if (curr_appl != app_new):
		return True
return False
```

Reply:
Taken care